crates/router/src/core/payments/routing.rs

```diff
@@ -232,7 +232,7 @@ pub fn make_dsl_input(
     };
 
     let payment_input = dsl_inputs::PaymentInput {
-        amount: payments_dsl_input.payment_intent.amount,
+        amount: payments_dsl_input.payment_attempt.get_total_amount(),
```

Comment:
@Aprabhat19  Is routing need total_amount or just order_amount? In other places dsl_inputs do we need total amount?

Reply:
Ideally we should be doing on net_amount right? @Aprabhat19

Reply:
Yes we would do it on total_amount, cause he could have a rule saying `amount`>1000, route it to `connector_b`. So total_amount would be a more accurate field for making such decisions